Remove explicit OpenSSL dependency

zachmargolis opened the pull request:
**Why**: Try to fix error loading openssl 2.2.0 when deployed

aduth commented:
Where all do we use OpenSSL? I see it explicitly required here:

https://github.com/18F/identity-idp-functions/blob/67f44016ca8c58ebaf46c969401da941a06eefd6/source/aws-ruby-sdk/encryption_helper.rb#L1

Which makes me wonder: Should we have it defined as a dependency in [`source/aws-ruby-sdk/Gemfile`](https://github.com/18F/identity-idp-functions/blob/master/source/aws-ruby-sdk/Gemfile)?

zachmargolis commented:
> Which makes me wonder: Should we have it defined as a dependency in [`source/aws-ruby-sdk/Gemfile`](https://github.com/18F/identity-idp-functions/blob/master/source/aws-ruby-sdk/Gemfile)? (And only in that `Gemfile`?)

So that Gemfile is basically ignored AFAIK unfortunately, the Gemfile layers don't work the way I thought they would

zachmargolis commented:
Updated to use plan B: removing an explicit dependency and using the packaged/built-in one with the Ruby, also changed to lazily require openssl similar to what I just did in https://github.com/18F/identity-idp-functions/pull/46
